app/src/main/java/com/github/se/wanderpals/model/repository/TripsRepository.kt

```diff
@@ -61,6 +63,204 @@ class TripsRepository(
+                  .document(suggestionId)
+                  .get()
+                  .await()
+          val firestoreSuggestion = documentSnapshot.toObject<FirestoreSuggestion>()
+          if (firestoreSuggestion != null) {
+            firestoreSuggestion.toSuggestion()
+          } else {
+            Log.e(
+                "TripsRepository",
+                "getSuggestionFromTrip: Not found Suggestion $suggestionId from trip $tripId.")
+            null
+          }
+        } catch (e: Exception) {
+          Log.e(
+              "TripsRepository",
+              "getSuggestionFromTrip: Error getting a Suggestion $suggestionId from trip $tripId.",
+              e)
+          null // error
+        }
+      }
+
+  /**
+   * Retrieves all suggestions associated with a specific trip. It iterates over all suggestion IDs
+   * stored within a trip document and fetches their corresponding suggestion objects.
+   *
+   * @param tripId The unique identifier of the trip.
+   * @return A list of `Suggestion` objects. Returns an empty list if the trip is not found, if
+   *   there are no suggestions associated with the trip, or in case of an error during data
+   *   retrieval.
+   */
+  suspend fun getAllSuggestionsFromTrip(tripId: String): List<Suggestion> =
+      withContext(dispatcher) {
+        try {
+          val trip = getTrip(tripId)
+
+          if (trip != null) {
+            val suggestionIds = trip.suggestions
+            suggestionIds.mapNotNull { suggestionId -> getSuggestionFromTrip(tripId, suggestionId) }
+          } else {
+            Log.e("TripsRepository", "getAllSuggestionsFromTrip: Trip not found with ID $tripId.")
+            emptyList()
+          }
+        } catch (e: Exception) {
+
+          Log.e(
+              "TripsRepository",
+              "getAllSuggestionsFromTrip: Error fetching Suggestion to trip $tripId.",
+              e)
+          emptyList()
+        }
+      }
+
+  /**
+   * Adds a new suggestion to a specified trip. This involves creating a unique identifier for the
+   * suggestion, converting the suggestion to a Firestore-compatible format, and updating the trip's
+   * document to include the new suggestion. If successful, the method also updates the trip
+   * document to include the newly added suggestion's ID in the list of suggestions.
+   *
+   * @param tripId The unique identifier of the trip to which the suggestion is being added.
+   * @param suggestion The `Suggestion` object to be added to the trip.
+   * @return `true` if the suggestion was added successfully, `false` otherwise. Errors during the
+   *   process are logged.
+   */
+  suspend fun addSuggestionToTrip(tripId: String, suggestion: Suggestion): Boolean =
+      withContext(dispatcher) {
+        try {
+          val uniqueID = UUID.randomUUID().toString()
+          val firebaseSuggestion =
+              FirestoreSuggestion.fromSuggestion(suggestion.copy(suggestionId = uniqueID))
```

Comment:
TripsRepository already knows who is creating the Suggestion, it should also modify the userId and username of the Suggestion when copying the suggestion.

Reply:
Great catch about the UserId, however for the username, this is something that the app, should know. As such the username will need to be specified by the caller. Thx!